genesis/utils/usda_bake.py

```diff
@@ -0,0 +1,127 @@
+import os
+import argparse
+import logging
+import asyncio
+
+os.environ["OMNI_KIT_ACCEPT_EULA"] = "yes"
+
+
+def omni_bootstrap(device=0, log_level="warning"):
+    import omni.kit_app
+
+    app = omni.kit_app.KitApp()
+    kit_dir = os.path.dirname(os.path.abspath(os.path.realpath(omni.kit_app.__file__)))
+    kit_path = os.path.join(kit_dir, "apps", "omni.app.empty.kit")
+    app_args = [
+        kit_path,
+        f"--/app/window/hideUi=True",
+        f"--/app/tokens/exe-path={kit_dir}",
+        f"--/app/enableStdoutOutput=False",  # Disable print outs (print_and_log) on extension startup information
+        f"--/app/runLoops/present/rateLimitFrequency=60",
+        f"--/app/vulkan=True",
+        f"--/app/asyncRendering=False",
+        f"--/app/python/interceptSysStdOutput=False",
+        f"--/app/python/logSysStdOutput=False",
+        f"--/app/settings/fabricDefaultStageFrameHistoryCount=3",
+        f"--/omni/log/level={log_level}",
+        f"--/log/file=",  # Empty string means no log file
+        f"--/log/level={log_level}",
+        f"--/renderer/activeGpu={device}",
+        f"--/renderer/enabled=rtx",
+        f"--/renderer/active=rtx",
+        f"--/renderer/multiGpu/maxGpuCount=1",  # Avoids unnecessary GPU context initialization
+        f"--no-window",
+        f"--portable",
+        "--enable",
+        "omni.usd",
+        "--enable",
+        "omni.kit.material.library",
+        "--enable",
+        "omni.kit.viewport.utility",
+        "--enable",
+        "omni.kit.viewport.rtx",
+        "--enable",
+        "omni.kit.usd.collect",
+        "--enable",
+        "omni.replicator.core",
+        "--enable",
+        "omni.mdl.distill_and_bake",
+    ]
+    app.startup(app_args)
+    app.update()  # important
+    return app
+
+
+def bake_usd_material(input_file, output_dir, usd_material_paths, device=0, log_level="error"):
+    # sys.stdout = open(os.devnull, "w")
```

Comment:
Remove comments. If you need to hide some "prints" for other libraries, either use
```python
from contextlib import redirect_stdout

 with open(os.devnull, 'w') as stdout, redirect_stdout(stdout):
     [...]
```
for Python-scope prints.

Or 
```python
from genesis.utils.misc import redirect_libc_stderr

with open(os.devnull, "w") as stderr, redirect_libc_stderr(stderr):
    [...]
```
for C++ scope stderr silencing.

Reply:
If you want to store the messages in a buffer and delay printing, you can do this:
```python
import io
from contextlib import redirect_stdout

_outputs = io.StringIO()

with redirect_stdout(_outputs):
    [...]

for line in _outputs.getvalue().splitlines():
    logger.debug(line)
```